src/app/BreedProcessor.ts

```diff
@@ -16,29 +16,34 @@ export class BreedProcessor {
     this.randomResultInterfaceFox = new FirstDice();
   }
 
-  processBreedPhase({ theHerd }: Player): void {
-    const wolf = this.getRandomResult(
-      this.randomResultInterfaceWolf.getRandomValue(),
-    );
-    const fox = this.getRandomResult(
-      this.randomResultInterfaceFox.getRandomValue(),
-    );
-    if (typeof fox !== 'object') {
-      if (typeof wolf !== 'object') {
-        this.breedAnimals(fox, wolf, theHerd);
+  processBreedPhase({ theHerd }: Player): [AnimalNames, AnimalNames] {
+    const wolfDiceResult = this.randomResultInterfaceWolf.getRandomValue();
+    const foxDiceResult = this.randomResultInterfaceFox.getRandomValue();
+    if (foxDiceResult !== AnimalNames.FOX) {
+      if (wolfDiceResult !== AnimalNames.WOLF) {
```

Comment:
There is a lot of ifs and elses, perhaps switch case? To minimize repeated code, perhaps cosider using a `const isAWolf = wolfDiceResult === AnimalNames.WOLF`? Or perhaps write a separete function ex. `handleDiceResult(isWolf, isFox, animal1, animal2)` where you could do all the logic.

Reply:
I will consider this and refactor, if I will find suitable solution, and/or i will get information from client what should happen in few specific situations.